src/app.js

```diff
@@ -17,6 +17,21 @@ const logger = require('../logs/logger');
 const app = express();
 require('express-async-errors');
 
+// Serve the Swagger JSON file
+// app.get('/swagger.json', (req, res) => {
+//   res.sendFile('swagger-output.json');
+// });
+
+// Serve Swagger UI static files
+app.use(
+  '/swagger-ui',
+  express.static(path.join(__dirname, 'node_modules/swagger-ui-dist'))
```

Comment:
`__dirname` points to the directory of this file, so `./src`. But `node_modules` is in the root folder above. So currently this is point ing to `/path/to/backend/repo/src/node_modules/swagger-ui-dist` which is not an existing file. 

You can either:

1.  Use `..` to point to the folder _above_ this one: `app.use("/swagger-ui", express.static(path.join(__dirname, '../node_modules/swagger-ui-dist')`
2.  The docs for `swagger-ui-dist` show that it exports a function `absolutePath` that we can use here instead: [link](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/installation.md#:~:text=The%20module%27s%20contents%20mirror%20the%20dist%20folder%20you%20see%20in%20the%20Git%20repository.%20The%20most%20useful%20file%20is%20swagger%2Dui%2Dbundle.js%2C%20which%20is%20a%20build%20of%20Swagger%20UI%20that%20includes%20all%20the%20code%20it%20needs%20to%20run%20in%20one%20file.%20The%20folder%20also%20has%20an%20index.html%20asset%2C%20to%20make%20it%20easy%20to%20serve%20Swagger%20UI%20like%20so%3A)

```
const pathToSwaggerUi = require('swagger-ui-dist').absolutePath()

app.use("/swagger", express.static(pathToSwaggerUi));
```

3\. `app.use("/swagger", express.static(path.dirname(require.resolve('swagger-ui-dist'))));` 

(`require.resolve` is built-in to Node and returns the main file of a particular package, `path.dirname` is also a built-in method and returns the path to a directory containing a given file)

Reply:
Thank you very much for your detailed explanation of this issue, I have changed according to point 2, commit 82d6c10a4f5e87b0caae768677210b10170f13ee

Reply:
As for the swagger documentation on my backend I have successful response but on localhost  by the path / swagger I see the example of swagger UI but not the actual routes to login/logout 
I don't manage to see as in your screenshot
  <img width="847" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/88336329/2447a018-9013-4487-96e3-2adfcd63690e">

Reply:
Akos , what do you think if we disable cors in app.js ? May be it will help me to see the endpoints but not swagger html as it is now happening? 
something like this `app.use(cors()); // Allow all origins `

Reply:
![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/2f1bf105-6aa5-4b09-a9b8-69562998b9be)

Sorry this is what I meant earlier, if you put in the path to the swagger json route `/swagger.json` into the search bar at the top; it will show the generated API docs instead of the example ones. I haven't had a chance yet but I'll look into if there's a way to by default show that instead of the example pet store endpoints. But I think as long as you document it (just a quick sentence in the readme or in a documentaion folder) so others know how to find it; this is okay to merge for now.

Reply:
Akos I am sorry that I ask again but if I use the path  /swagger.json  I see the json file but not swagger one. May be I do something wrong? 

<img width="589" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/88336329/c13e5d8f-0c8c-4520-a196-93d35ae0295d">

<img width="534" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/88336329/310a1339-d6c2-496b-ab4e-3eb0d5f2b5df">